src/libsystemd-network/dhcp6-option.c

```diff
@@ -167,6 +168,40 @@ int dhcp6_option_append_fqdn(uint8_t **buf, size_t *buflen, const char *fqdn) {
         return r;
 }
 
+int dhcp6_option_append_vendor_class(uint8_t **buf, size_t *buflen, char **vendor_class) {
+        _cleanup_free_ uint8_t *p = NULL;
+        uint32_t enterprise_identifier;
+        size_t total, offset;
+        char **s;
+
+        assert_return(buf && *buf && buflen && vendor_class, -EINVAL);
```

Comment:
use assert_return() only in user-facing validation checks.

assert() is how we detect our own programming errors. and assert_return() is a slightly more friendly version that we use to detect programming errors of others. The former logs and aborts, the latter logs and returns from the function. 

This means: assert_return() should pretty much be used only in functions prefixed with `sd_`, as those are the exported, user facing ones. but functions without sd_ prefixes are used only in our own code and thus should use assert() for checking arguments.